src/libraries/System.IO.Compression/ref/System.IO.Compression.cs

```diff
@@ -131,18 +131,30 @@ public enum ZipArchiveMode
         Create = 1,
         Update = 2,
     }
-    public sealed class ZLibCompressionOptions
+    public sealed partial class ZLibCompressionOptions
     {
-        public int CompressionLevel { get; set; }
-        public ZLibCompressionStrategy CompressionStrategy { get; set; }
+        public ZLibCompressionOptions() { }
+        public int CompressionLevel { get { throw null; } set { } }
+        public System.IO.Compression.ZLibCompressionStrategy CompressionStrategy { get { throw null; } set { } }
     }
     public enum ZLibCompressionStrategy
     {
         Default = 0,
         Filtered = 1,
         HuffmanOnly = 2,
         RunLengthEncoding = 3,
-        Fixed = 4
+        Fixed = 4,
+    }
+    public partial class ZLibException : System.IO.IOException, System.Runtime.Serialization.ISerializable
```

Comment:
Seems like this was intentionally omitted https://github.com/dotnet/runtime/blob/4da708a7060d238354d79823e58839c1d92f7fd1/src/libraries/System.IO.Compression/src/CompatibilitySuppressions.xml#L3-L7

Perhaps you can configure GenAPI so that it will exclude it?

Reply:
Let me understand how it works first.

Reply:
I'll open an issue to keep investigating later https://github.com/dotnet/runtime/issues/112451

I'll jump to work on the m-p issue.

Reply:
I added a comment to https://github.com/dotnet/runtime/issues/112451#issuecomment-2654662416 with how to do this.  Your call if you want to do it in this PR or not.

Reply:
Ah thanks. I think I should do it in this PR.

Reply:
Pushed the changes. Nice to see that after creating the txt file and adding that property to the csproj, the ZLibException is no longer getting auto-added to the ref cs when running the `dotnet msbuild /t:GenerateReferenceAssemblySource ` command.